src/tasks/initiate_new_auction.ts

```diff
@@ -84,6 +90,30 @@ const initiateAuction: () => void = () => {
       );
 
       console.log("Using EasyAuction deployed to:", easyAuction.address);
+      if (
+        taskArgs.allowListManager !=
+        "0x0000000000000000000000000000000000000000"
+      ) {
+        const allowListManager = await hardhatRuntime.ethers.getContractAt(
+          "AllowListVerifier",
+          taskArgs.allowListManager,
+        );
+        const interfaceSignatureOfIsAllowed = allowListManager.interface
+          .getSighash("isAllowed(address,uint256,bytes)")
+          .substring(2);
+        const cutByteCodeFromContract = (
+          await allowListManager.provider.getCode(allowListManager.address)
+        ).substring(0, 500); // Byte code is cut to 500 byte to make sure that we only search for the interface bytes at the beginning to avoid false negatives
+
+        if (cutByteCodeFromContract.includes(interfaceSignatureOfIsAllowed)) {
```

Comment:
I'd add a comment explaining what's going on here, this is hard to understand without extra details.

```suggestion
        // The bytecode of a smart contract contains a table with the signatures
        // of all functions that are directly defined in the contract as public
        // or external. The function table appears at the start. Note that there
        // could be false negatives (e.g., proxies) and false positives (very
        // large contracts or, unlikely, in case other unrelated parts of the
        // bytecode matches the four signature bytes).
        if (cutByteCodeFromContract.includes(interfaceSignatureOfIsAllowed)) {
```